src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java

```diff
@@ -92,9 +92,9 @@
             this.origin = SCMHeadOrigin.DEFAULT;
         } else {
             // if the forked repo name differs from the upstream repo name
-            this.origin = pr.getHead().getRepository().getName().equalsIgnoreCase(pr.getBase().getRepository().getName())
+            this.origin = pr.getBase().getRepository().getName().equalsIgnoreCase(sourceRepo)
```

Comment:
Your test and the proposed code looks good to me but I am not sure to understand why in such situation `pr.getBase().getRepository()` isn't null ?